mb_DiscordRichPresence.cs

```diff
@@ -176,29 +173,19 @@ private void UpdatePresence(string artist, string track, string album, bool play
             // NOTE(yui): this is very ugly
             string year = null;
 
-            if (yearStr.Length > 0)
-            {
-                try
-                {
-                    year = DateTime.Parse(yearStr).Year.ToString();
-                }
-                catch (FormatException)
-                {
-                    if (yearStr.Length == 4)
-                    {
-                        year = DateTime.ParseExact(yearStr, "yyyy", null).Year.ToString();
-                    }
-                }
+            if (yearStr.Length > 0 && config.showYear)
+            {
+                DateTime result;
+
+                if (DateTime.TryParse(yearStr, out result))
+                    year = result.Year.ToString();
+                else
+                    if (yearStr.Length == 4)
+                        if (DateTime.TryParseExact(yearStr, "yyyy", null, System.Globalization.DateTimeStyles.None, out result))
+                            year = result.Year.ToString();
             }
 
-            if (year != null && config.showYear)
-            {
-                presence.largeImageText = $"{album} ({year})";
-            }
-            else
-            {
-                presence.largeImageText = album;
-            }
+            presence.largeImageText = $"{album}" + ( year != null ? $" ({year})" : "" );
```

Comment:
Readd `&& config.showYear`

Reply:
I pushed the change,

`presence.largeImageText = $"{album}" + ( year != null ? $" ({year})" : "" );`
to 
`presence.largeImageText = $"{album}" + ( (year != null && config.showYear) ? $" ({year})" : "" );`

i think this is what you mean.